crates/luminol/src/app/top_bar.rs

```diff
@@ -0,0 +1,322 @@
+                        let mut cmd = luminol_term::CommandBuilder::new("game");
+                        cmd.cwd(
+                            update_state
+                                .filesystem
+                                .project_path()
+                                .expect("project not loaded"),
+                        );
+
+                        luminol_ui::windows::console::Window::new(cmd)
+                    });
+
+                    match result {
+                        Ok(w) => update_state.edit_windows.add_window(w),
+                        Err(e) => update_state.toasts.error(format!(
+                            "error starting game (tried steamshim.exe and then game.exe): {e}"
+                        )),
+                    }
+                }
+
+                if ui.button("Terminal").clicked() {
+                    #[cfg(windows)]
+                    let shell = "powershell";
+                    #[cfg(unix)]
+                    let shell = std::env::var("SHELL").unwrap_or_else(|_| "bash".to_string());
+                    let mut cmd = luminol_term::CommandBuilder::new(shell);
+                    cmd.cwd(
+                        update_state
+                            .filesystem
+                            .project_path()
+                            .expect("project not loaded"),
+                    );
+
+                    match luminol_ui::windows::console::Window::new(cmd) {
+                        Ok(w) => update_state.edit_windows.add_window(w),
+                        Err(e) => update_state
+                            .toasts
+                            .error(format!("error starting shell: {e}")),
+                    }
+                }
+            });
+        }
+
+        ui.separator();
+
+        ui.label("Brush:");
+
+        for brush in luminol_core::Pencil::iter() {
+            ui.selectable_value(&mut update_state.toolbar.pencil, brush, brush.to_string());
+        }
+
+        if open_project {
+            self.load_project_promise = Some(poll_promise::Promise::spawn_async(
+                luminol_filesystem::host::FileSystem::from_pile_picker(),
+            ));
+        }
+
+        if save_project {
+            if let Some(config) = update_state.project_config {
+                update_state.toasts.info("Saving project...");
+                match update_state.data.save(update_state.filesystem, config) {
+                    Ok(_) => update_state.toasts.info("Saved project sucessfully!"),
+                    Err(e) => update_state.toasts.error(e.to_string()),
+                }
+            }
+        }
+
+        if let Some(p) = self.load_project_promise.take() {
+            match p.try_take() {
+                Ok(Ok(host)) => {
+                    if let Err(why) = update_state.data.load(
+                        update_state.filesystem,
+                        // TODO code jank
+                        update_state.project_config.as_mut().unwrap(),
```

Comment:
The application panics here with "called `Option::unwrap()` on a `None` value" when loading a project using the File menu in the top bar. Project loading using the button in the Get Started tab works, however!

Reply:
I thought I fixed this, but apparently not? I'll look into it again